src/screens/NotificationScreen.js

```diff
@@ -1,23 +1,141 @@
 import React from "react";
-import { SafeAreaView, Text, StyleSheet } from "react-native";
+import { SafeAreaView, StyleSheet, View, Text } from "react-native";
 
-import colors from "../config/colors";
+import { scale, moderateScale, verticalScale } from "../config/scaling";
+import colorConfig from "../config/colors";
+import screenConfig from "../config/screen";
+
+import ContentItem from "../component/Notification/ContentItem";
 
 const NotificationScreen = ({ navigation }) => {
+  const notification = [
+    {
+      status: "TODAY",
+      name: "Jasper Matthews1",
+      vaccineName: "COVID-19 Vaccine",
+      date: "April 15th",
+    },
+    {
+      status: "TODAY",
+      name: "Jasper Matthews2",
+      vaccineName: "COVID-19 Vaccine2",
+      date: "April 15th",
+    },
+    {
+      status: "UPCOMING",
+      name: "Jasper Matthews3",
+      vaccineName: "COVID-19 Vaccine3",
+      date: "April 15th",
+    },
+    {
+      status: "OVERDUE",
+      name: "Jasper Matthews4",
+      vaccineName: "COVID-19 Vaccine4",
+      date: "April 15th",
+    },
+  ];
+
+  const statusUpcoming = () => {
+    return (
+      <View>
+        {notification
+          .filter((item) => item.status === "UPCOMING")
+          .map((item, index) => {
+            return (
+              <ContentItem
+                key={index}
+                name={item.name}
+                vaccineName={item.vaccineName}
+                date={item.date}
+              />
+            );
+          })}
+      </View>
+    );
+  };
+
+  const statusOverdue = () => {
+    return (
+      <View>
+        {notification
+          .filter((item) => item.status === "OVERDUE")
+          .map((item, index) => {
+            return (
+              <ContentItem
+                key={index}
+                name={item.name}
+                vaccineName={item.vaccineName}
+                date={item.date}
+              />
+            );
+          })}
+      </View>
+    );
+  };
+
+  const statusToday = () => {
+    return (
+      <View>
+        {notification
+          .filter((item) => item.status === "TODAY")
+          .map((item, index) => {
+            return (
+              <ContentItem
+                key={index}
+                name={item.name}
+                vaccineName={item.vaccineName}
+                date={item.date}
+              />
+            );
+          })}
+      </View>
+    );
+  };
+
```

Comment:
Functions named statusToday, statusUpcoming and statusOverdue are duplicates in terms of the logic of function. Make a one function, pass the status to filter the array

Reply:
When we need to use date to determine the status, we do need 3 separate functions. I delete this. Ignore this review.